### PR TITLE
feat(minifier): implement block stmt support for `StatementFusion`

### PR DESCRIPTION
**Pull request by camc314**

### CODE REVIEW

**Comment by camc314** [comment]
* **#6422** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6422?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camc314 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_minifier/src/ast_passes/statement_fusion.rs [hunk 1]
```diff
@@ -312,14 +331,14 @@ mod test {
...
-            "function f(a) { if (COND) {  { a,b; let otherVariable = 1; } } }",
-        );
+        // test(
+        //     "function f(a) { if (COND) { a; { b; let a = 1; } } }",
+        //     "function f(a) { if (COND) { { a,b; let a$jscomp$1 = 1; } } }",
+        // );
+        // test(
+        //     "function f(a) { if (COND) { a; { b; let otherVariable = 1; } } }",
+        //     "function f(a) { if (COND) {  { a,b; let otherVariable = 1; } } }",
+        // );
```
Comment (camc314) [inline crates/oxc_minifier/src/ast_passes/statement_fusion.rs:341]: Closure checks whether `isASTNormalized` (here) https://github.com/google/closure-compiler/blob/081f46cb61c8852ec6a863db18e6a9d91693ce23/src/com/google/javascript/jscomp/StatementFusion.java#L92

oxc_minifier doesn't have this concept yet, so i guess its ok to leave these commented out for now?

Comment (Boshen) [reply crates/oxc_minifier/src/ast_passes/statement_fusion.rs:341]: Yes. I don't even know what "normalized" ast means.

**Comment by Boshen** [comment]
### Merge activity

* **Oct 10, 10:40 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Comment by Boshen** [comment]
Now I recall why they were commented out: https://github.com/oxc-project/monitor-oxc/actions/runs/11277037252/job/31362358789

@camc314 this is harder than it seems.

**Comment by camc314** [comment]
😬 will look

**Comment by camc314** [comment]
think this is a codegen bug
any `'foo' in bar` should be wrapped in parenthasizedexpr if they are in a for loop else it fails to parse.
will look further...

**Comment by camc314** [comment]
Fixed in #6425

**Pull request merged**